### PR TITLE
Base pipeline and raster.py changes for generating isoscapes

### PR DESCRIPTION
**Pull request by VimWizardVersace**

### CODE REVIEW

File: raster.py [hunk 1]
```diff
@@ -272,6 +276,56 @@ def get_data_at_coords(dataset: AmazonGeoTiff, x: float, y: float, month: int) -
   else:
     return value
 
+def get_predictions_at_each_pixel(
+    model: tf.keras.Model,
+    feature_transformer: ColumnTransformer,
+    geotiffs: dict[str, AmazonGeoTiff],
+    bounds: Bounds):
```
Comment (erickzul) [inline raster.py:283]: Could we please add a method description here and specify in it what these parameters are?

File: raster.py [hunk 2]
```diff
@@ -317,5 +371,113 @@ def cellulose_isoscape_geotiff() -> AmazonGeoTiff:
...
+  "d13C_cel_var": d13C_var_geotiff,
+  "ordinary_kriging_linear_d18O_predicted_mean" : krig_means_isoscape_geotiff,
+  "ordinary_kriging_linear_d18O_predicted_variance" : krig_variances_isoscape_geotiff
+}
+
+def generate_isoscapes_from_variational_model(
+    output_geotiff_id: str,
+    model: tf.keras.Model,
+    feature_transformer: ColumnTransformer,
+    required_geotiffs: List[str],
```
Comment (erickzul) [inline raster.py:466]: Do we need to specify here that they need to be in the order of the model input.
Also, is there a way to get this from the feature_transformer or some metadata of the model so we don't need to "keep track" of the order of the columns?

**Review by erickzul — CHANGES_REQUESTED** [review]
We need to add tests for the new changes to raster.py. See hypothesis_test.ipynb and dataset_test.ipynb as examples.

File: raster.py (hunk 2 above)
Comment (VimWizardVersace) [reply raster.py:466]: I couldn't find a to store the column training info as part of the model. In a follow up set of PRs I just train them and load them in alphabetical order.

Comment (VimWizardVersace) [reply raster.py:466]: Also, FeatureTransformer does appear to have the columns in order they were trained, but I don't think that's a guarantee. I'd rather be responsible for sorting it ourselves.

**Comment by VimWizardVersace** [comment]
> We need to add tests for the new changes to raster.py. See hypothesis_test.ipynb and dataset_test.ipynb as examples.

Added a small test to raster_test.ipynb

File: raster.py (hunk 1 above)
Comment (erickzul) [reply raster.py:283]: Did we resolve this?

**Review by erickzul — APPROVED** [review]
LGTM pending resolving a comment.

**Pull request merged**